packages/mdc-checkbox/_mixins.scss

```diff
@@ -0,0 +1,414 @@
+//
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import "@material/animation/functions";
+@import "@material/ripple/mixins";
+@import "@material/rtl/mixins";
+@import "./functions";
+@import "./keyframes";
+@import "./variables";
+
+//
+// Public
+//
+
+@mixin mdc-checkbox-unmarked-stroke-color($color) {
+  @include mdc-checkbox-unmarked-background-selector-enabled_ {
+    @include mdc-theme-prop(border-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-marked-stroke-color($color) {
+  @include mdc-checkbox-marked-background-selector-enabled_ {
+    @include mdc-theme-prop(border-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-marked-container-fill-color($color) {
+  @include mdc-checkbox-marked-background-selector-enabled_ {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-ink-color($color) {
+  .mdc-checkbox__checkmark__path {
+    // !important is needed here because a stroke must be set as an attribute on the SVG in order
+    // for line animation to work properly.
+    @include mdc-theme-prop(stroke, $color, $important: true);
+  }
+
+  .mdc-checkbox__mixedmark {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-focus-indicator-color($color) {
+  // The ::before element is used as a focus indicator for the checkbox
+  .mdc-checkbox__background::before {
+    @include mdc-theme-prop(background-color, $color, $edgeOptOut: true);
+  }
+}
+
+@mixin mdc-checkbox-ripple($ripple-config) {
+  $ripple-config: map-merge((opacity: $mdc-checkbox-ripple-opacity), $ripple-config);
+
+  @include mdc-ripple-bg(map-merge((pseudo: "::before"), $ripple-config));
+  @include mdc-ripple-fg(map-merge((pseudo: "::after"), $ripple-config));
+
+  &::before,
+  &::after {
+    // This needs to be included immediately after mdc-ripple-bg and mdc-ripple-fg
+    // to override their default dimensions.
+    @include mdc-checkbox-ripple-size_;
+  }
+}
+
+@mixin mdc-checkbox-mark-animation(
```

Comment:
This was probably discussed and I missed it, but are we concerned at all that you need to invoke multiple mixins with the same parameters?

Also, I'm confused by unmarked-container-color, since there's no individual mixin for that and I thought the idea was it's transparent?

Reply:
I'm definitely concerned about it. Animating from one color to another requires specifying both the start color and end color, so I'm not sure how we could work around this. Suggestions welcome!

Reply:
To reduce confusion, I removed the non-animation stroke/container mixins. There's now just one mixin that does all 4: `mdc-checkbox-stroke-and-container-fill-color()`. It also optionally generates the `@keyframes` needed by the JavaScript component.